lib/src/browser/disposable_browser.dart

```diff
@@ -60,11 +240,8 @@ class Disposable extends disposable_common.Disposable {
   void _subscribeToEvent(EventTarget eventTarget, String event,
       EventListener callback, bool useCapture) {
     eventTarget.addEventListener(event, callback, useCapture);
-
-    var disposable = new disposable_common.InternalDisposable(() {
+    _disposable.manageDisposer(() {
       eventTarget.removeEventListener(event, callback, useCapture);
     });
-
-    disposable_common.addInternalDisposable(this, disposable);
```

Comment:
I changed the logic to create and manage a `Disposer` so we no longer need to jump through the hoops provided by `addInternalDisposable`.